New Setting Customize for keys And New Style por Activitys

RetrogisusDEV opened the pull request:
What more?

Julow commented on res/values/styles.xml:
```diff
@@ -22,4 +22,20 @@
+  <style name="appTheme" parent="@android:style/Theme.Material.Light">
+    <item name="android:textStyle">bold</item>
+    <item name="android:colorAccent">#ff2424ff</item>
+    <item name="android:colorBackground">#ffeeeeff</item>
```
These colors don't look natural on my device. Why not use the default colors by not specifying them ?

RetrogisusDEV replied:
After meditating, I thought about the number of devices with different designs, and well, I thought about the topic more for new devices (Holo: I'm still alive), so I'm going to leave only the basic modifiers and leave the rest as default, for right, what api is your device?

RetrogisusDEV replied:
The point of my changes is to make the keyboard with more universal modifications, for different people, I love flat designs, but others don't, that's why I have to think more global and less local

Julow reviewed (commented):
There's a lot of good ideas, thanks!

You were based on a very old commit so I rebased your work on the latest version and pushed to your branch.
I've also fixed the missing imports and re-indented the code and xml files.

To get the same on your side, use these commands:
```
git fetch https://github.com/RetrogisusDEV/Unexpected-Keyboard master
git reset --hard FETCH_HEAD
```

RetrogisusDEV commented:
> There's a lot of good ideas, thanks!
> 
> You were based on a very old commit so I rebased your work on the latest version and pushed to your branch.
> I've also fixed the missing imports and re-indented the code and xml files.
> 
> To get the same on your side, use these commands:
> ```
> git fetch https://github.com/RetrogisusDEV/Unexpected-Keyboard master
> git reset --hard FETCH_HEAD
> ```

Damn... well, I uploaded it stupidly because I wrote everything on the phone and well the format went on vacation...

 Actually I did not base it on the previous work, I did everything from scratch, improving the integration and making it more efficient than the previous one, after the customize key thing, I have seen that when leaving the keyboard for a while it clicks and you have to disable it to to work again (the whole app) so I have to verify

 Based on a proposal I read, I am trying to use an accessibility service (more specifically, the accessibility button in the navigation bar to invoke the keyboard at any time...

 did I do the service?  : Yeah
 works?  : No

 I don't know how to insert the code without causing the service to self-disable (or explode)

Julow commented on res/xml/settings.xml:
```diff
@@ -46,5 +46,8 @@
     <juloo.keyboard2.prefs.IntSlideBarPreference android:key="key_horizontal_space" android:title="@string/pref_key_horizontal_space" android:summary="%sdp" android:defaultValue="2" min="0" max="8"/>
+    <CheckBoxPreference android:key="border_config" android:title="@string/pref_border_config" android:summary="@string/pref_border_config_summary" android:defaultValue="false"/>
+    <juloo.keyboard2.prefs.IntSlideBarPreference android:key="border_radius" android:title="@string/pref_corners_radius" android:summary="%sdp" android:summaryOff="@string/pref_border_options_disabled" android:defaultValue="0" min="0" max="15" android:dependency="border_config"/>
+    <juloo.keyboard2.prefs.IntSlideBarPreference android:key="border_line_size" android:title="@string/pref_border_width" android:summaryOff="@string/pref_border_options_disabled" android:summary="%sdp" android:defaultValue="0" min="0" max="5" android:dependency="border_config"/>
```
`1dp` is a lot, perhaps this option should be a `juloo.keyboard2.prefs.SlideBarPreference` that works with floating point numbers.

Julow commented on srcs/juloo.keyboard2/Config.java:
```diff
@@ -147,6 +150,10 @@ public void refresh(Resources res)
     keyActivatedOpacity = _prefs.getInt("key_activated_opacity", 100) * 255 / 100;
+  	// keyboard border settings
+		borderConfig = _prefs.getBoolean("border_config", false);
+		borderRadius = _prefs.getInt("border_radius", 0);
```
The preference contain numbers in the `dp` unit but `borderRadius` is in `px` unit. It must be converted like this:
```suggestion
		borderRadius = get_dip_pref(dm, "border_radius", 0);
```
Same for borderLineSize.

Julow commented on srcs/juloo.keyboard2/Config.java:
```diff
@@ -59,6 +61,7 @@ public final class Config
   public boolean autocapitalisation;
   public boolean switch_input_immediate;
   public boolean pin_entry_enabled;
+  public boolean borderConfig;
```
This option might seem like a hack but is a foundation for a future "Custom theme" option. Ideally it would work by creating a `Theme` object and not by adding conditions in the rendering code. Is anyone interested in working on that?

Julow commented:
I merged the light/dark theme switch in a55506e. I did not include the customized text styles because they look non-native. Simpler is better.

Julow commented:
I've made several changes and pushed to your branch. I'll merge!

RetrogisusDEV commented:
@Julow Pipipi I haven't had an internet connection for a while, that's why I disappear and reappear 😭

RetrogisusDEV commented:
Well I'm going to give you a summary of what I was doing 

**Custom Theme Adjustment / Default**

Based on your suggestion, I made a adjustment to change all the settings for customizations or already default settings, and with the intention of adding other settings that work with this setting [Almost Ready] 

**Keyboard Background Adjustment Hex / RGB**

 A setting that is activated with the previous one, to be able to change the color of the keyboard background, using 3 sliders to change the values in base r . G. B , and then convert it into hex [the keyboard does not want to be multicolored, that's why it doesn't work xD ] 

**accessibility service to invoke the keyboard**

 because for something I read for there, it made me want (I already had ) to do and to be able to invoke the keyboard where be, and well, accessibility is the most shareable thing that occurred to me, which is using the accessibility button (the one who is a man in stand position [the joke takes fun if it were a 3D model]) and invoke the keyboard or close it, useful for games or Visual novels (they pass them to me, after a while I find out that it's porn and I'm playing that like it's swimming in public xD) that require the keyboard at a certain moment but sometimes the keyboard takes some things [the joke is that, if I did the service, but when you activate it, the button appears in the corner for 200 ms, and says: ehhh! a black! - Insert Mr. Hair Animation. and it disappears, despite the fact that keyboard or anything is affected by bug or crasho xD ] 

**fix a custom border error**

 well yes, said setting seemed something simple, but it is the devil disguised as a code, because when turning off the phone or Restart it, you realize that when trying to use the keyboard it does not start, and if you try to open the app it doesn't start either [it's really weird, I haven't achieved anything, but the keyboard restarts in a loop every 3 seconds or less, and yes, I have 1 week without the keyboard, I'm using the backup {see 1.23.x} it fixes reinstalling it from 0, but when you turn off the cel occurs again lol ]

Julow commented:
All that seems very nice :) 
I've made several changes to your branch then merged. Make sure to start from the last version of the `master` branch of the right repository.

Invoking the keyboard is a popular feature request and I'm happy that you're looking into it!

For the crash, could you get the logs with `adb logcat` ?

RetrogisusDEV commented:
The problem is that uhh, I modified some things in Windows and now my phone doesn't detect it properly and uhh, I think ADB won't work for me 😕, I'll have to repair Windows

Julow commented:
Can you try the apps mentioned in the middle of this page ? https://f-droid.org/docs/Getting_logcat_messages_after_crash/
I couldn't reproduce the crash on my side.

RetrogisusDEV commented:
> Can you try the apps mentioned in the middle of this page ? https://f-droid.org/docs/Getting_logcat_messages_after_crash/
> I couldn't reproduce the crash on my side.

Tell you something funny? The joke is that there was never an error, the PC did recognize my phone (but not as pipipi storage) and I spent 1 hour looking at the ADB LOGCAT and well, an error as such never appeared, just a " Warning" of "The application forcibly closed the keyboard" and that's it xd

RetrogisusDEV commented:
advance: I already made an option to change the colors of the keyboard [Keys/Background], separately, and depending on the color of the keys, (being some value greater than 150) it changes the color of the letters from black to white

[option for change color Default(Theme) / Custom (RGB value to HEX)]

![Screenshot_20240211-175710.png](https://github.com/Julow/Unexpected-Keyboard/assets/99091223/262b4888-5437-4c56-be6a-3c1be5e2a20d)

[Result]

![Screenshot_20240211-175531.png](https://github.com/Julow/Unexpected-Keyboard/assets/99091223/42c4f1b1-6958-4767-ba92-7d81f7aea8ac)

Julow commented:
Interesting, I'd be glad to review your PR.

RetrogisusDEV commented:
hey @Julow what function is it that updates the keyboard, I mean, it updates the color data, since I don't know what it is and well, I've been lazy these days (damn Touhou, I even see it at food)

Julow commented:
The keyboard view can be redrawn by first [refreshing the global configuration](https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Config.java#L103) then calling the [`reset` method](https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Keyboard2View.java#L110).

However, when changing the theme, the view must be re-created like here: https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Keyboard2.java#L225C38-L225C50 (the `inflate_view` function is defined at the end).

RetrogisusDEV commented:
@Julow  In a while I'm going to start making the PR of what I've done and send it to you, to revive the repo xD
